Client/Application/AppDelegate.swift

```diff
@@ -135,6 +142,52 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
         SDWebImageDownloader.sharedDownloader().setValue(firefoxUA, forHTTPHeaderField: "User-Agent")
     }
+
+    func application(application: UIApplication, handleActionWithIdentifier identifier: String?, forLocalNotification notification: UILocalNotification, completionHandler: () -> Void) {
+        if let actionId = identifier {
+            if let action = SentTabAction(rawValue: actionId) {
+                switch(action) {
+                case .View:
+                    viewURLInNewTab(notification)
+                    break
+                case .Bookmark:
+                    addBookmark(notification)
+                    break
+                case .ReadingList:
```

Comment:
Was this a UX request? I'm not sure if it makes sense to have an Add To Reading List button in the notification banner. @darrinhenein @tecgirl what do you think? The interaction is a bit odd. Because you receive the notification in a different app. Then tap 'Add to Bookmarks / RL', which does that, but also activates Firefox and does not actually show the page. You have to see this in action I think.